src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -500,9 +508,13 @@ namespace winrt::TerminalApp::implementation
                 wc.command = action;
                 wc.weight = weight;
 
-                // If we're in Tab Switcher mode, we want to fall back to Index
-                // order, not Command Name alphabetical order.
-                wc.orderByTabIndex = _currentMode == CommandPaletteMode::TabSwitcherMode;
+                // I hate this but this adds an "inOrderCounter" to allow commands with
+                // the same names and weights to be ordered in the order they were pushed
+                // onto the heap. This specifically helps out tab actions in the Tab Switcher,
+                // since it's quite common to have tab titles be the same.
+                uint32_t counter = 0;
```

Comment:
nit: could we just declare `counter` outside the for loop, and just do a `wc.inOrderCounter = counter++;` each time, instead of re-calculating the index?